communication/communications.py

```diff
@@ -77,8 +78,30 @@ def _write(self, sig, opcode, arg, seqNo):
         Sends the code to the Arduino
         """
         checksum = self.create_checksum(arg, opcode)
-        opcode_string = "%d%d%03d%d\r" % (sig, opcode, arg, checksum)
+        opcode_string = "%d%d%03d%d%d\r" % (sig, opcode, arg, checksum, seqNo)
```

Comment:
This looks fine, maybe extract it out into a function so it can be used again on line 101?